src/test/bridges/example/Example.t.sol

```diff
@@ -1,76 +1,55 @@
+        vm.label(address(exampleBridge), "Example Bridge");
+        vm.prank(MULTI_SIG);
+        ROLLUP_PROCESSOR.setSupportedBridge(address(exampleBridge), 100000);
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
     }
 
-    function testErrorCodes() public {
-        AztecTypes.AztecAsset memory empty;
-
-        address callerAddress = address(bytes20(uint160(uint256(keccak256("non-rollup-processor-address")))));
-
-        vm.prank(callerAddress);
+    function testErrorCodes(address _callerAddress) public {
+        vm.assume(_callerAddress != address(ROLLUP_PROCESSOR));
+        vm.prank(_callerAddress);
         vm.expectRevert(ErrorLib.InvalidCaller.selector);
-        exampleBridge.convert(empty, empty, empty, empty, 0, 0, 0, address(0));
+        exampleBridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
     }
 
-    function testExampleBridge() public {
-        uint256 depositAmount = 15000;
+    function testExampleBridge(uint256 _depositAmount) public {
+        uint256 depositAmount = bound(_depositAmount, 2, type(uint96).max);
+        AztecTypes.AztecAsset memory daiAsset = getRealAztecAsset(address(DAI));
+
         // Mint the depositAmount of Dai to rollupProcessor
-        deal(address(DAI), address(rollupProcessor), depositAmount);
+        deal(address(DAI), address(ROLLUP_PROCESSOR), depositAmount);
 
-        AztecTypes.AztecAsset memory empty;
-        AztecTypes.AztecAsset memory inputAsset = AztecTypes.AztecAsset({
-            id: 1,
-            erc20Address: address(DAI),
-            assetType: AztecTypes.AztecAssetType.ERC20
-        });
-        AztecTypes.AztecAsset memory outputAsset = AztecTypes.AztecAsset({
-            id: 1,
-            erc20Address: address(DAI),
-            assetType: AztecTypes.AztecAssetType.ERC20
-        });
+        uint256 bridgeId = encodeBridgeId(id, daiAsset, emptyAsset, daiAsset, emptyAsset, 0);
+        vm.expectEmit(true, true, false, true);
+        emit DefiBridgeProcessed(bridgeId, getNextNonce(), depositAmount, depositAmount, 0, true, "");
+        sendDefiRollup(bridgeId, depositAmount);
 
-        // Disabling linting errors here to show return variables
-        // solhint-disable-next-line
-        (uint256 outputValueA, uint256 outputValueB, bool isAsync) = rollupProcessor.convert(
-            address(exampleBridge),
-            inputAsset,
-            empty,
-            outputAsset,
-            empty,
-            depositAmount,
-            1,
-            0
-        );
+        assertEq(depositAmount, DAI.balanceOf(address(ROLLUP_PROCESSOR)), "Balances must match");
 
-        uint256 rollupDai = DAI.balanceOf(address(rollupProcessor));
+        uint256 secondDeposit = depositAmount / 2;
+        vm.expectEmit(true, true, false, true);
+        emit DefiBridgeProcessed(bridgeId, getNextNonce(), secondDeposit, secondDeposit, 0, true, "");
```

Comment:
I think the vm.expectEmit(...) is one of the most confusing cheatcodes  so I would link [foundry book](https://book.getfoundry.sh/forge/cheatcodes.html) so that grantees have a reference. Ideally I would comment basically each line in this example. If I was a grantee and I didn't understand rollups much (which is most of the grantees) I would be just super confused what's going on here. Having a well documented example test would help a lot.